### PR TITLE
fix(deps): frozen installation fails during ugprade workflow

### PR DESCRIPTION
**Pull request by iliapolo**
Fixes https://github.com/projen/projen/issues/800.

The problem was that projen itself can change `package.json` files due to its own requirements. 

> For example https://github.com/cdklabs/cdk-triggers/blob/main/.projenrc.js#L6

Which causes the `package.json` file to change after running `yarn install` - and a subsequent frozen installation would fail. 

This PR uses projen itself to install the dependencies and update the lock file, so this happens after all `package.json` mutations have happened. 

---

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/upgrade-dependencies.ts [hunk 1]
```diff
@@ -118,7 +118,11 @@ export class UpgradeDependencies extends Component {
     }
 
     task.exec(ncuCommand.join(' '));
-    task.exec(this._project.package.renderInstallCommand(false));
+
+    // we use projen to install the dependencies and thus update the lock file,
+    // so we need to make it think its not running inside CI.
+    task.exec(`CI=0 ${this._project.projenCommand}`);
```
Comment (eladb) [inline src/upgrade-dependencies.ts:124]:
```suggestion
    task.exec(`${this._project.projenCommand}`, {
      env: { CI: '0' }
    });
```

Comment (iliapolo) [reply src/upgrade-dependencies.ts:124]: Cool

File: src/upgrade-dependencies.ts [hunk 2]
```diff
@@ -118,7 +118,11 @@ export class UpgradeDependencies extends Component {
     }
 
     task.exec(ncuCommand.join(' '));
-    task.exec(this._project.package.renderInstallCommand(false));
```
Comment (eladb) [inline src/upgrade-dependencies.ts:121]: Does that mean we can make this private again?

Comment (iliapolo) [reply src/upgrade-dependencies.ts:121]: Yes

**Review by eladb — APPROVED** [review]

**Pull request merged**